core/src/main/java/org/elasticsearch/plugins/InstallPluginCommand.java

```diff
@@ -160,13 +162,15 @@ private Path download(String pluginId, Path tmpDir) throws Exception {
             return downloadZipAndChecksum(url, tmpDir);
         }
 
-        // now try as maven coordinates, a valid URL would only have a single colon
-        String[] coordinates = pluginId.split(":");
-        if (coordinates.length == 3) {
-            String mavenUrl = String.format(Locale.ROOT, "https://repo1.maven.org/maven2/%1$s/%2$s/%3$s/%2$s-%3$s.zip",
-                coordinates[0].replace(".", "/") /* groupId */, coordinates[1] /* artifactId */, coordinates[2] /* version */);
-            terminal.println("-> Downloading " + pluginId + " from maven central");
-            return downloadZipAndChecksum(mavenUrl, tmpDir);
+        // now try as maven coordinates
+        if (false == isURL(pluginId)) {
```

Comment:
I don't like doing this "try a url and on error do something". Can we instead add an extra condition to the clause, so in addition to having 3 coordinates separated by colon, it does not contain `/`? All URLs will have a slash. Then we don't have to worry about this code trapping a malformed URL and attempting as maven coordinates, and the URL handling below can handle malformed urls.